lib-injection/auto_inject.rb

```diff
@@ -0,0 +1,19 @@
+if ENV['DDTRACE_AUTOINJECT'] && !ENV['skip_autoinject']
+  if system 'skip_autoinject=true bundle show ddtrace'
+    puts 'ddtrace already installed...'
+  else
+    puts 'ddtrace is not installed... Perform auto-injection...'
+
+    if system 'skip_autoinject=true bundle add ddtrace'
+      puts 'ddtrace added to bundle...'
```

Comment:
When `bundle add` fails, it can leave the Gemfile modified but the Gemfile.lock untouched. (e.g. `bundle install && bundle install --deployment && bundle add nokogiri ; bundle exec irb` will fail to execute `bundle exec irb`)

This can cause errors when the application tries to run `bundle exec` in order to start.

We should either completely succeed with the auto-injection, or not leave  any changes in the system.